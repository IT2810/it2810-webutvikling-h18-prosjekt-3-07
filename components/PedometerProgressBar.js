import React from 'react';

import {
  View,
  StyleSheet,
  Dimensions,
  Text,
} from 'react-native';
 
import ProgressBarAnimated from 'react-native-progress-bar-animated';
 
export default class PedometerProgressBar extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      progress: props.steps,
      progressWithOnComplete: 0,
    }
  }
  componentWillReceiveProps(newProp) {
      this.setState({progress: (newProp.steps/100)})
   }

   checkGoalComplete = (progress) => {
     if (progress >= 100) {
       return 100
     }else {
      return progress
     }
   }
   
   

  render() {
    const barWidth = Dimensions.get('screen').width - 30;
    return (
       
      <View style={styles.container}>
        <View>
          <ProgressBarAnimated
            width={barWidth}
            value={this.checkGoalComplete(this.state.progress)}
            backgroundColor='#4d94ff'
            backgroundColorOnComplete='#47d147'
          />
        </View>
        <Text style={styles.text}>{this.state.progress}
          <Text>% completed</Text>
        </Text>
      </View>
    );
  }
}
 
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFF',
    marginTop: 50,
    padding: 15,
  },
  text: {
    paddingTop: 10,
    alignSelf: 'center',
    color: 'gray',
    fontSize: 20,
    fontWeight: 'normal',
  },
});