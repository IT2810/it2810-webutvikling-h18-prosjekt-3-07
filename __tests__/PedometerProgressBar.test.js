import React from "react";
import renderer from "react-test-renderer";
import PedometerProgressBar from "../components/PedometerProgressBar";

it("renders correctly", () => {
  const tree = renderer.create(<PedometerProgressBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
