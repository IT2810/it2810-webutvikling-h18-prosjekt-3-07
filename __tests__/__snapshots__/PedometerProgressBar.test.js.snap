// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "marginTop": 50,
      "padding": 15,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "#C8CCCE",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 15,
          "width": 720,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(77, 148, 255, 1)",
            "borderRadius": 6,
            "height": 13,
            "width": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "gray",
        "fontSize": 20,
        "fontWeight": "normal",
        "paddingTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      % completed
    </Text>
  </Text>
</View>
`;
